akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -605,6 +605,14 @@ import scala.util.control.NonFatal
     // call has no effect and therefore does the right thing: nothing.
     try {
       inputs.foreach(_.onInternalError(reason))
+      // fail all sources
+      logics.foreach { logic ⇒
+        // source
+        if (logic.inCount == 0 && logic.outCount > 0) {
```

Comment:
Also, I think the original comment is still valid. All sink stages or stages where `onUpstreamFailure` doesn't propagate will still only have `postStop` called but not `onDownstreamFinish`.